### PR TITLE
Extract command parsing

### PR DESCRIPTION
**Pull request by MeBrei**
This extracts command parsing code that will be shared by the issue and diff app

### CODE REVIEW

File: rad/prelude/cmd-parsing.rad [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+(def parse-failure
+ (fn [error help]
+   (put-str! error)
+   (put-str! help)
+   (exit! 1)
+ ))
+
+ (def missing-arg-error
+   (fn [arg cmd]
+     (string-append "Missing argument \"" (show arg) "\" for command \"" cmd "\"")))
```
Comment (jkarni) [inline rad/prelude/cmd-parsing.rad]: Is this and the next function indented too much?

File: rad/prelude/cmd-parsing.rad [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    (fn [v]
+      (match v
+        [cmd] (parse-failure (string-append "Missing arguments for command " cmd) help)
+        (/cons cmd (/cons 'c-n 'os)) (if (member? c-n valid-opts)
+                                       (parse-failure (missing-arg-error chain-name cmd) help)
+                                       [:just {chain-name c-n
+                                               opts    (opts-map os)}])
+        _ :nothing))))
+
+(def /cmd-2
```
Comment (jkarni) [inline rad/prelude/cmd-parsing.rad]: Can you add a docstring to this and any other exported function?

File: rad/prelude/cmd-parsing.rad (hunk 1 above)
Comment (MeBrei) [reply rad/prelude/cmd-parsing.rad]: Fixed the indentations.

File: rad/prelude/cmd-parsing.rad (hunk 2 above)
Comment (MeBrei) [reply rad/prelude/cmd-parsing.rad]: Added docstrings. Let me know if they should be more detailed.

**Pull request merged**